### PR TITLE
Remove the use of ledgerDbCurrentValues in the ThreadNet tests

### PR DESCRIPTION
**Pull request by dnadales**
We now make use of a range query that fetches the whole UTxO from the given `DiskLedgerView`.

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [x] New tests are added if needed and existing tests are updated
    - [x] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [x] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [x] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

**Review by Jasagredo — APPROVED** [review]
This looks good to me.

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs [hunk 1]
```diff
@@ -636,13 +644,147 @@ data LedgerBackingStoreValueHandle m l = LedgerBackingStoreValueHandle
...
+          HD.UedsIns       -> 0
+          HD.UedsInsAndDel -> 0
+
+    -- INVARIANT: nrequested > 0
+    --
+    -- (1) if we reached the end of the store, then simply yield the given diff
+    --     applied to the given values
+    -- (2) otherwise, the readset must be non-empty, since 'rqCount' is positive
+    -- (3) remove the greatest read key
+    -- (4) remove all diff elements that are >= the greatest read key
```
Comment (jorisdral) [inline ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs:726]: If I'm reading the code correctly, I think it is doing the opposite of (4)

**Review by jorisdral — APPROVED** [review]
LGTM -- Only some minor comments

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs [hunk 2]
```diff
@@ -636,13 +644,147 @@ data LedgerBackingStoreValueHandle m l = LedgerBackingStoreValueHandle
 
 type LedgerBackingStore' m blk = LedgerBackingStore m (ExtLedgerState blk)
 
+mkDiskLedgerView ::
+     (GetTip (l EmptyMK), IOLike m, TableStuff l)
+  => (LedgerBackingStoreValueHandle m l, LedgerDB l, m ())
```
[the remark this answers is not in the document]
Comment (dnadales) [reply ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs:649]: Do you think that would make the code more readable? Sometimes having all the components in one line makes reading faster. But I don't mind either way.

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs (hunk 1 above)
Comment (dnadales) [reply ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs:726]: Are you referring to this line? 

```
HD.UtxoDiff $ Map.filterWithKey (\dk _dv -> dk < k) ds
```

Here we're filtering *out* all the differences whose key is >= the greatest read key (`k` in this case).

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs [hunk 3]
```diff
@@ -636,13 +644,147 @@ data LedgerBackingStoreValueHandle m l = LedgerBackingStoreValueHandle
...
+      -- ^ TODO: what is this?
+      -> ApplyMapKind DiffMK   k v
+      -- ^ TODO: what is this?
+      -> ApplyMapKind ValuesMK k v
+      -- ^ TODO: what is this?
+      -> ApplyMapKind ValuesMK k v
+    doFixupReadResult
+      nrequested
+      (ApplyDiffMK (HD.UtxoDiff ds))
+      (ApplyValuesMK (HD.UtxoValues vs)) = -- TODO: it's hard to see where the fundef begins
```
Comment (dnadales) [inline ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs:760]: I forgot to delete this. I wanted to ask where that equal sign should be. To me, it's hard to see where the function definition begins. Could this be an alternative? 

```haskell
    doFixupReadResult
      nrequested
      (ApplyDiffMK (HD.UtxoDiff ds))
      (ApplyValuesMK (HD.UtxoValues vs)) 
    = 
        let includingAllKeys              =
              HD.forwardValues (HD.UtxoValues vs) (HD.UtxoDiff ds)
            definitelyNoMoreToFetch       = Map.size vs < nrequested
            assertDefinitelyNoMoreToFetch =
              if Map.size vs < nrequested
              then id
              else error $ "Size of values " <> show (Map.size vs) <> ", nrequested " <> show nrequested
        in ...
```

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs [hunk 4]
```diff
@@ -636,13 +644,147 @@ data LedgerBackingStoreValueHandle m l = LedgerBackingStoreValueHandle
...
+    -- queries could guide us here.
+    doFixupReadResult ::
+         Ord k
+      => Int
+      -- ^ TODO: what is this?
+      -> ApplyMapKind DiffMK   k v
+      -- ^ TODO: what is this?
+      -> ApplyMapKind ValuesMK k v
+      -- ^ TODO: what is this?
+      -> ApplyMapKind ValuesMK k v
```
[the remark this answers is not in the document]
Comment (dnadales) [reply ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs:756]: Yes. Thanks for the reminder :pray:

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jorisdral) [reply ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs:648]: I don't mind -- I remembered I had read something about this in the style guide

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs (hunk 3 above)
Comment (jorisdral) [reply ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs]: Certainly, or maybe have the arguments on one line and pattern match/extract `ds` and `vs` somewhere else?

**Pull request merged**